tests/test_nrnhines.py

```diff
@@ -42,13 +43,40 @@ def test_NeuroM_section_to_NRN_compartment_paths():
 
 
 def test_point_to_section_end():
-    cell = test_module.get_NRN_cell(SIMPLE)
-    assert_equal(test_module.point_to_section_end(cell.icell.all, [-8, 10, 0]),
+    cell = tested.get_NRN_cell(SIMPLE)
+    assert_equal(tested.point_to_section_end(cell.icell.all, [-8, 10, 0]),
                  6)
 
-    assert_equal(test_module.point_to_section_end(cell.icell.all, [-8, 10, 2]),
+    assert_equal(tested.point_to_section_end(cell.icell.all, [-8, 10, 2]),
                  None)
 
-    assert_equal(test_module.point_to_section_end(cell.icell.all, [-8, 10, 2],
-                                                  atol=10),
+    assert_equal(tested.point_to_section_end(cell.icell.all, [-8, 10, 2],
+                                             atol=10),
                  3)
+
+
+def _to_be_isolated(morphology_path, point):
+    """Convert a point position to NEURON section index and return cell name and id.
+
+    Args:
+        morphology_path (Path): path to morphology
+        point (list/ndarray): 3d point
+    """
+    cell = tested.get_NRN_cell(morphology_path)
+    return tested.point_to_section_end(cell.icell.all, point)
+
+
+def _isolated(morph_data):
+    """Serves as a decorator to isolate convert_point_to_isc, but work with multiprocessing."""
+    return tested.isolate(_to_be_isolated)(*morph_data)
+
+
+def test_isolate():
+    n_workers = 4
+    with tested.NestedPool(processes=n_workers) as pool:
```

Comment:
Why not use `maxtasksperchild=1` here and skip the `NestedPool` altogether?

Reply:
the thing is what runs inside _isolated is a multiprocess.Pool already, so we need NestedPool to make this Pool non demonic. This is the whole trick, which I don't fully comprehend tbh. if you have a simpler solution, it's cool! This is originally adapted from BluePyOpt/BluePyMM of @wvangeit . Do you need that?

Reply:
I was just pointed here as an example of dask memory issues… with the multiprocessing given as the cause.

I'm just wondering… if the outer pool already uses `maxtasksperchild=1`, then the inner pool that is used to "isolate" the function seems a bit superfluous. If this is covered by the tests, I can give it a try tomorrow.

I'm also curious why `dask.bag` was used originally rather than `multiprocessing`. Would keep the dependency trees a little smaller.

Reply:
@asanin-epfl and benoit wanted to use dask here, but I was more in favor of multiprocessing indeed, for this simple thing. With @adrien-berchet , we have been fighting some issues with dask, and we are begining to win (not with bag though). Maybe we can setup some dask channel on slack to share our fights ;)